src/lib/y2network/sysconfig/connection_config_writer.rb

```diff
@@ -35,7 +35,7 @@ class ConnectionConfigWriter
       # @param conn [Y2Network::ConnectionConfig::Base] Connection configuration to write
       # @param old_conn [Y2Network::ConnectionConfig::Base,nil] Connection configuration to write
       def write(conn, old_conn = nil)
-        return if conn == old_conn
+        return if old_conn && conn == old_conn
```

Comment:
this looks like mistake of `operator==` which should not crash when comparison to nil, so please fix  it there.